core/src/main/scala/cats/Traverse.scala

```diff
@@ -30,6 +30,12 @@ import simulacrum.typeclass
     U.TC.traverse(fa)(a => U.subst(f(a)))(this)
 
   /**
+   * A traverse followed by flattening the inner result.
+   */
+  def traverseM[G[_], A, B](fa: F[A])(f: A => G[F[B]])(implicit G: Applicative[G], F: FlatMap[F]): G[F[B]] =
+    G.map(traverse(fa)(f))(F.flatten)
```

Comment:
I was not really sure about the `M`. `traverseA` seems a bit strange because we need `FlatMap`. `traverseF` could also be confusing with the `flatMapF` and `mapF` on (some) monad transformers.

Maybe we can call it `traverseFlatten` ?